### ISSUE TITLE
Java JDT.LS configuration error

### ISSUE THREAD

**Issue by jqno** [kabouzeid/nvim-lspinstall#65]
I've installed and configured the Java language server, and when I open a Java file, the language server starts up correctly, but I get a warning in the first line that says `<filename>.java is a non-project file, only syntax errors are reported`. This is due to the fact that the `-data` parameter in the `jdtls.sh` is not configured correctly. It gets the name of the workspace, which is good (it prevents that data files from different projects get mixed up), but it needs an absolute path to work, not a relative path.

If we change this line: https://github.com/kabouzeid/nvim-lspinstall/blob/8f7ce91fad7affa08a9a0d47fddfa37923fd3f4e/lua/lspinstall/servers/java.lua#L134
to something like `-data "$(pwd)/workspaces/\$WORKSPACE" \\` everything works fine.

It would be easy for me to create a PR for this, however, I'm not sure if the directory I chose here is the correct one. It seems like it should be in `~/.cache/nvim` somewhere, but I'm not sufficiently up-to-date in Neovim's directory usage that I can say for certain. And if so, I also don't know what the best way would be to put that directory into the script without hard-coding it.

What do you think? I'm happy to provide a PR for this.

**Comment by jqno** [comment]
Also, now that I'm here: the script calls a command called `die` in two places, but my machine (a freshly installed Ubuntu 21.04) doesn't recognise it, and I can't find it via `apt` either. Googling it seems to turn up a lot of false positives about processes dying and whatnot. What do I need to install to make that work?

**Comment by kabouzeid** [comment]
@jqno I copied parts of the script from kotlin lang server. Didn't realize `die` is a cmd local to the script, thank (see: https://github.com/fwcd/kotlin-language-server/blob/58966bc18dd5bf82fdfcf7a2f44bab3622f6907a/server/src/test/resources/additionalWorkspace/gradlew#L56).

`$WORKSPACE` should already be an absolute path https://github.com/kabouzeid/nvim-lspinstall/blob/8f7ce91fad7affa08a9a0d47fddfa37923fd3f4e/lua/lspinstall/servers/java.lua#L34-L39

I also just tested it to confirm

```vim
:lua print(require "lspconfig/util".path.join { require"lspinstall/util".install_path("java"), 'workspace', 'test' })
```
prints:
```
/Users/karim/.local/share/nvim/lspinstall/java/test
```

**Comment by jqno** [comment]
Ah, my bad. I came from a setup without lspinstaller where the full path was handled by the script instead of the config object.
Since I use https://github.com/mfussenegger/nvim-jdtls, I still need to explicitly pass a workspace, and I forgot to update the `cmd` in the lua config.

Thanks for the quick and helpful response!

**Comment by kabouzeid** [comment]
So now you've removed the `cmd` key from your config?

**Comment by jqno** [comment]
I'm using nvim-jdtls, so I can't use lspconfig's setup function for Java. I have to give it a config object, so I still have to define a `cmd` key. I've kept what I have, but changed the workspace parameter so that it now contains a full path.

I was unable to figure out how to get the config object including the `cmd` key from lspinstall, and use that as a basis. I'm still finding my way around Lua and Neovim's APIs 😅. So I figured I'd just hard code it for now and improve it later.

Is there a good way to get that config object, and/or a nice/official way to get the location of a language server for a given language? Right now I'm doing this: https://github.com/jqno/dotfiles/blob/4142375cd0961e1442f42128289acb7bc43c06aa/nvim/lua/filetypes/java.lua#L29
